src/ott/tools/k_means.py

```diff
@@ -151,14 +151,14 @@ def body_fn(
     centroid_dists = candidate_dists[best_ix]
 
     return KPPState(
-        key=next_key, centroids=centroids, centroid_dists=centroid_dists
+        rng=next_key, centroids=centroids, centroid_dists=centroid_dists
     )
 
   if n_local_trials is None:
     n_local_trials = 2 + int(math.log(k))
   assert n_local_trials > 0, n_local_trials
 
-  state = init_fn(geom, key)
+  state = init_fn(geom, rng)
```

Comment:
here `rng` was not changed during the last `split` contrary to the code before the change, this might lead to a bug.

Reply:
This should be fine, as apart from the `init_fn`, the `rng` is not used anywhere.